__tests__/components/FormInputEditable.spec.ts

```diff
@@ -0,0 +1,121 @@
+
+interface FormInputEditableProps {
+    model?: any;
+    label: string;
+    value: string;
+    rules?: string;
+    onEdit?: (value: string) => void;
+}
+
+describe('components/FormInputEditable', () => {
+    const getFormInputEditableWrapper = (props?: FormInputEditableProps) => {
+        const localVue = createLocalVue();
+        localVue.component('ValidationProvider', ValidationProvider);
+        localVue.use(Vuex);
+        localVue.use(VueI18n);
+        const store = new Vuex.Store({
+            modules: {},
+        });
+        const options = {
+            localVue,
+            i18n,
+            store,
+            propsData: { ...props },
+            stubs: ['Tooltip'],
+        };
+        const wrapper = mount(FormInputEditable, options);
+        return wrapper;
+    };
+
+    test('renders component', () => {
+        // Arrange + Act:
+        const props = {
+            value: 'myAccountName',
+            label: 'myLabel',
+        };
+        const wrapper = getFormInputEditableWrapper(props);
+
+        // Assert:
+        expect(wrapper.find('span.accountName').text()).toBe(props.value);
+        expect(wrapper.find('div.label').text()).toBe(props.label + ':');
+    });
+
+    const testEditingSuccess = async (valueProp?: string) => {
+        // Arrange:
+        const props = {
+            value: valueProp,
+            model: {},
+            label: 'myLabel',
+            rules: '',
+            onEdit: jest.fn(),
+        };
+        const newAccountName = 'myNewAccountName';
+        const wrapper = getFormInputEditableWrapper(props);
+        const component = wrapper.vm as FormInputEditableTs;
+
+        // Act:
+        await wrapper.find('button').trigger('click');
+        await wrapper.find('input').setValue(newAccountName);
+        component.finishEdition();
+
+        // Assert:
+        expect(component.value).toBe(newAccountName);
+        expect(props.onEdit).toHaveBeenCalledWith(newAccountName);
+    };
+
+    test('complete editing with success when value is not empty', async () => {
+        testEditingSuccess('myAccountName');
+    });
+
+    test('complete editing with success when value is empty', async () => {
+        testEditingSuccess();
+    });
+
+    test('cancel editing', async () => {
+        // Arrange:
+        const props = {
+            value: 'myAccountName',
+            model: {},
+            label: 'myLabel',
+            rules: '',
+            onEdit: jest.fn(),
+        };
+        const newAccountName = 'myNewAccountName';
+        const wrapper = getFormInputEditableWrapper(props);
+        const component = wrapper.vm as FormInputEditableTs;
+
+        // Act:
+        await wrapper.find('button').trigger('click');
+        await wrapper.find('input').setValue(newAccountName);
+        component.cancelEdition();
+
+        // Assert:
+        expect(component.editing).toBe(false);
+        expect(props.onEdit).not.toHaveBeenCalled();
+    });
+});
```

Comment:
consider test where component.editing is true

Reply:
added 👍